src/crush/CrushWrapper.cc

```diff
@@ -893,7 +893,9 @@ pair<string,string> CrushWrapper::get_immediate_parent(int id, int *_ret)
         string parent_id = name_map[b->id];
         string parent_bucket_type = type_map[b->type];
         loc = make_pair(parent_bucket_type, parent_id);
-        ret = 0;
+        if (_ret)
+          *_ret = 0;
+        return loc;
```

Comment:
if we do this, then we should get rid of int ret and loc at the top, move them into the inner scope, and make the end of the method unconditionally set *_ret = -ENOENT and return pair<string,string>().